core/proofs/sector.hpp

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef CPP_FILECOIN_CORE_PROOFS_SECTOR_HPP
+#define CPP_FILECOIN_CORE_PROOFS_SECTOR_HPP
+
+#include "primitives/cid/cid.hpp"
+#include "primitives/types.hpp"
+#include "proofs/proofs.hpp"
+
+namespace fc::proofs::sector {
+  using common::Buffer;
+  using primitives::ActorId;
+  using primitives::ChainEpoch;
+  using primitives::DealId;
+  using primitives::SectorNumber;
+
+  struct SectorId {
+    ActorId miner;
+    SectorNumber sector;
+  };
+
+  enum class RegisteredProof : int64_t {
+    WinStackedDRG32GiBSeal = 1,
+    WinStackedDRG32GiBPoSt,
+    StackedDRG32GiBSeal,
+    StackedDRG32GiBPoSt,
+    StackedDRG1KiBSeal,
+    StackedDRG1KiBPoSt,
+    StackedDRG16MiBSeal,
+    StackedDRG16MiBPoSt,
+    StackedDRG256MiBSeal,
+    StackedDRG256MiBPoSt,
+    StackedDRG1GiBSeal,
+    StackedDRG1GiBPoSt,
+  };
+
+  using SealRandomness = Randomness;
+
+  using InteractiveRandomness = Randomness;
+
+  struct SealProof {
+    Buffer proof;
+  };
+
+  struct OnChainSealVerifyInfo {
```

Comment:
Structures are partly commented in specs-actors, please leave comment for structures where it is possible.